test/unit/gateways/cyber_source_rest_test.rb

```diff
@@ -233,6 +233,166 @@ def test_url_building
     assert_equal "#{@gateway.class.test_url}/pts/v2/action", @gateway.send(:url, 'action')
   end
 
+  def test_successful_credit_card_purchase_single_request_ignore_avs
```

Comment:
💬 Comment:

well this is not the intended usage of the `Mocha#with`, even when you can use the block as an assertion it self is kind of weird to have assertions inside of it, I think you have two options to improve here:

1) Follow the standard way to check the request body like [this](https://github.com/activemerchant/active_merchant/blob/master/test/unit/gateways/cyber_source_rest_test.rb#L188)

2) use the with as an assertion but just checking one thing: like this

```ruby
 @gateway.expects(:ssl_post).with do |_host, request_body|
      json_body = JSON.parse(request_body)
      json_body['processingInformation']['authorizationOptions']['ignoreAvsResult'] == 'true'
 end.returns(successful_purchase_response)
```

My take => I prefer (1) for clarity and consistency